examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls

```diff
@@ -11,7 +11,12 @@ global class B2CDeliverySample implements sfdc_checkout.CartShippingCharges {
         try {
             // In the Winter '21 release there should be two delivery groups per cart.
             // We need to get the ID of the cart delivery group in order to create the cart delivery group methods.
-            Id cartDeliveryGroupId = [SELECT CartDeliveryGroupId FROM CartItem WHERE CartId = :cartId][0].CartDeliveryGroupId;
+            List<CartItem> cartItems = new List<CartItem>([SELECT CartDeliveryGroupId FROM CartItem WHERE CartId = :cartId]);
+
+            Set<Id> cartDeliveryGroupsIDs = new Set<Id>();
+            for(CartItem curCartItem : cartItems){
```

Comment:
You should not do this. For large cart, this loop is very expensive. You should query from cartDeliveryGroup table directly to get cartDeliveryGroup IDs
SELECT Id from CartDeliveryGroup WHERE CartId = :cartId

Reply:
@wei-liu-salesforce fixed (the flip side, we will now add methods to delivery groups even those that possibly have no items, but that is less an issue for demo code, as those are limited to 20)